statsmodels/nonparametric/tests/test_kernel_density.py

```diff
@@ -414,3 +415,13 @@ def test_efficient_user_specified_bw(self):
                                                           randomize=False,
                                                           n_sub=100))
         npt.assert_equal(dens.bw, bw_user)
+
+
+@pytest.mark.parametrize("kernel", ["biw", "cos", "epa", "gau",
+                                    "tri", "triw", "uni"])
+def test_all_kernels(kernel, reset_randomstate):
+    data = np.random.normal(size=200)
+    x_grid = np.linspace(min(data), max(data), 200)
+    density = sm.nonparametric.KDEUnivariate(data)
+    density.fit(kernel="gau", fft=False)
+    assert isinstance(density.evaluate(x_grid), np.ndarray)
```

Comment:
I think you can assert that is allclose to list comprehension
`[density.evaluate(x_i for xi in x_grid]`

Reply:
It is just there to make sure that something is tested.  This is mostly a smoke check to ensure that the bug was actually fixed.

Reply:
But it doesn't verify that it is a correct fix

Reply:
It does. It raised the original error on this line.

Reply:
In my reading np.all shouldn't be needed. The filtering is element wise.
So the values should be scalar or 0-d.

If there were another bug, then np.all might propagate it and then the values wouldn't be the same in list comprehension versus x_grid